Add PostSync Hook option to change Argo CD duration reported (#26805)
DataDog/documentation: content/en/continuous_delivery/deployments/argocd.md

## Setup

> The setup below uses the [Webhook notification service][5] of Argo CD in order to send notifications to Datadog.

Reviewer: ```suggestion
The setup below uses the [Webhook notification service][5] of Argo CD to send notifications to Datadog.
```

   application is configured to deploy more than one service, see [Tag an Argo CD application deploying multiple services](#tag-an-argo-cd-application-deploying-multiple-services).
4. You can use the `dd_customtags` annotation to optionally add custom tags to the deployment executions generated for this Argo CD application.
   The value should be set to a comma-separated list of tags, structured as `key:value` pairs.

After you have subscribed your Argo CD application by adding the annotations above, new deployments of the application will start to appear in Datadog.

The [Recommended Setup](#recommended-setup) section below contains recommended actions to improve the monitoring reported in CD Visibility.

## Recommended Setup

### Change duration to wait for resources health
The duration reported in deployment events matches the sync duration in Argo CD. However, the sync duration generally represents the time spent by Argo CD to sync the Git repository state and the Kubernetes cluster state.
This means that what happens after the sync (for example, the time spent by the Kubernetes resources to start up) is not included in the duration.

> To change the duration reported to wait until the configured resources have started up and reached a healthy state, you can add a [PostSync Hook][19] to the resources that your Argo CD application monitors.

Reviewer 2: ```suggestion
To change the duration reported to wait until the configured resources have started up and reached a healthy state, add a new no-op resource monitored by your Argo CD application, with a [PostSync Hook][19] annotation.
```

    dd_k8s_cluster: example-cluster
```

## Change duration to wait for resources health
The duration reported in deployment events matches the sync duration in Argo CD. However, the sync duration generally represents the time spent by Argo CD to sync the Git repository state and the Kubernetes cluster state.
This means that what happens after the sync (for example, the time spent by the Kubernetes resources to start up), is not included in the duration.

To change the duration reported to wait until the configured resources have started up and reached a healthy state, you can add a [PostSync Hook][19] to the resources that your Argo CD application monitors.
The PostSync Hook will run after all the resources have reached a Healthy state, and the Argo CD sync will wait on the PostSync Hook result to update the application status as Healthy.

> Below is represented an example of a PostSync Hook Job that runs a simple `echo` command.

Reviewer 2: People have to add a noop container in order to get the hook to be sent after the application is healthy? Is there any other alternative? I feel it's odd to just say that this is an example without explaining what else they could do (or not do)

If the command has been correctly run, deployments contain Git metadata from the application repository instead of the configuration repository. Also, the deployment executions view now contains a new **Pipeline** tab representing the related CI pipeline trace.

## Visualize deployments in Datadog

> The [**Deployments**][6] and [**Executions**][7] pages populate with data after a deployment is executed. For more information, see [Search and Manage][9] and [CD Visibility Explorer][10].

Reviewer 2: ```suggestion
The [**Deployments**][6] and [**Executions**][7] pages populate with data after a deployment has finished. For more information, see [Search and Manage][9] and [CD Visibility Explorer][10].
```

I think that executed is too broad, it does not say that data comes only when the deployment has finished.
